### PR TITLE
Spark: Add table sort-order to reserved table properties

### PR DESCRIPTION
**Pull request by flyrain**
Unit test is added. The integration test is blocked by https://github.com/apache/iceberg/issues/2382.

### CODE REVIEW

**Comment by flyrain** [comment]
Can you take a look? @aokolnychyi , @RussellSpitzer , @karuppayya

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java [hunk 1]
```diff
@@ -880,4 +885,59 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
...
+    @Override
+    public String bucket(String sourceName, int sourceId, int numBuckets,
+                         org.apache.iceberg.SortDirection direction, NullOrder nullOrder) {
+      return String.format("bucket(%s, %s) %s %s", numBuckets, sourceName, direction, nullOrder);
+    }
+
+    @Override
+    public String truncate(String sourceName, int sourceId, int width,
+                           org.apache.iceberg.SortDirection direction, NullOrder nullOrder) {
+      return String.format("bucket(%s, %s) %s %s", sourceName, width, direction, nullOrder);
```
Comment (RussellSpitzer) [inline spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java:910]: Truncate?

Comment (RussellSpitzer) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java:910]: The tests don't cover all these cases, maybe we should add them in?

**Comment by aokolnychyi** [comment]
I think the primary focus of this PR is to show the current sort order as a table property rather than to add sort-order to the list of reserved properties. We will need a separate PR to prohibit modifying reserved properties but that's separate.

**Comment by aokolnychyi** [comment]
Let me take a look at this PR.

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java (hunk 1 above)
Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java:910]: Should we create a separate suite for artificial table properties like sort-order, current-snapshot-id?

**Comment by aokolnychyi** [comment]
This looks good apart from the point found by @RussellSpitzer. I'd be great to add a separate suite for all artificial properties we add in `SparkTable` where we could test different expression types.

**Comment by aokolnychyi** [comment]
Thanks for doing this, @flyrain!

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java (hunk 1 above)
Comment (flyrain) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java]: Fixed the comments in the new commits.

**Comment by flyrain** [comment]
Thanks @aokolnychyi and @RussellSpitzer for the review. I've resolved the comments in the new commits. Please take a look. Besides, user can use the command `ALTER TABLE local.db.student SET TBLPROPERTIES ('sort-order'='id asc')` to set the a `sort-order` properties. It is harmless since it won't take effect. But it is confusing. We should throw an error if user does it. We can include the change in another PR.

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java (hunk 1 above)
Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java]: Looks correct to me.

**Review by aokolnychyi — APPROVED** [review]

**Comment by aokolnychyi** [comment]
@flyrain, would you be interested to submit a follow-up PR to prohibit changes like you mentioned?

**Comment by flyrain** [comment]
@aokolnychyi, will submit a follow-up PR later.

**Comment by aokolnychyi** [comment]
Thanks for the PR, @flyrain! Thanks for reviewing, @RussellSpitzer!

**Pull request merged**